### PR TITLE
[699] - [BugTask] Notification red dot must be gone after the user click the notification bell button and notification must be gone if the user open the task notification slider.

### PR DESCRIPTION
**Pull request by rogeliojohnoliverio**
## Issue Link
- https://app.asana.com/0/1203011167276287/1203283041831699/f

## Definition of Done
- [x] After user tap/click the notification bell button, red dot will be automatically gone.
- [x] If user tap/click the notification bell button it will open the task slider and notification tab will be gone and only open the task slider.

## Notes
- Please run `php artisan migrate` again as new column `is_seen` is added for notifications table

## Pre-condition
- Go to a project then go to boards
- Assign a task to someone
- Click the notification bell, then the notification itself

## Expected Output
- After user tap/click the notification bell button, red dot will be automatically gone.
- If user tap/click the notification bell button it will open the task slider and notification tab will be gone and only open the task slider.

## Screenshots/Recordings

https://user-images.githubusercontent.com/110364637/199429620-f17a213d-d1a1-45c7-aa3c-5b1a08b379f1.mp4

### CODE REVIEW

File: api/database/migrations/2022_11_02_071347_add_seen_at_column.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+   * Reverse the migrations.
+   *
+   * @return void
+   */
+  public function down()
+  {
+    Schema::table('notifications', function (Blueprint $table) {
+      //
+    });
+  }
```
Comment (abduljalilpalala) [inline api/database/migrations/2022_11_02_071347_add_seen_at_column.php]: Implement the rollback of the migration file.

File: client/src/components/molecules/TaskList/index.tsx [hunk 2]
```diff
@@ -127,19 +130,10 @@ const TaskList: React.FC<Props> = (props): JSX.Element => {
...
-  const onSearch = (e: any): void => {
-    const value = e.target.value
-    const isEnter = e.key === 'Enter' || e.keyCode === 13
 
-    if (!isEnter) return
-    filterMembersName(parseInt(id as string), value)
-  }
+  useEffect(() => {
+    filterMembersName(parseInt(id as string), search)
+  }, [debouncedSearch])
```
Comment (abduljalilpalala) [inline client/src/components/molecules/TaskList/index.tsx:136]: What is the default type of `id`? If it was previously an integer then no need to parse.

File: client/src/components/organisms/TaskSlider/index.tsx [hunk 3]
```diff
@@ -130,19 +133,11 @@ const TaskSlider: FC<Props> = (props): JSX.Element => {
...
-    const value = e.target.value
-    const isEnter = e.key === 'Enter' || e.keyCode === 13
 
-    if (!isEnter) return
-    filterMembersName(parseInt(id as string), value)
-  }
+  useEffect(() => {
+    filterMembersName(parseInt(id as string), search)
+  }, [debouncedSearch])
+
```
Comment (abduljalilpalala) [inline client/src/components/organisms/TaskSlider/index.tsx]: Refer to my previous comment.

File: client/src/components/organisms/TaskSlider/index.tsx [hunk 4]
```diff
@@ -130,19 +133,11 @@ const TaskSlider: FC<Props> = (props): JSX.Element => {
...
   const handleSetName = (e: any) => {
     const keyCode = e.which || e.keyCode
```
Comment (abduljalilpalala) [inline client/src/components/organisms/TaskSlider/index.tsx]: Replace `any` with it's appropriate type.

File: client/src/hooks/notificationMethods.ts [hunk 5]
```diff
@@ -7,26 +7,38 @@ import { getSections, setProjectID } from '~/redux/section/sectionSlice'
 
 export const useNotificationMethods = () => {
   const [notifications, setNotifications] = useState([])
+  const [hasNotification, setHasNotification] = useState(false)
```
Comment (abduljalilpalala) [inline client/src/hooks/notificationMethods.ts]:
```suggestion
  const [hasNotification, setHasNotification] = useState<boolean>(false)
```

**Review by abduljalilpalala — COMMENTED** [review]
Please check comments.

File: client/src/components/molecules/TaskList/index.tsx (hunk 2 above)
Comment (rogeliojohnoliverio) [reply client/src/components/molecules/TaskList/index.tsx:136]: this is from query , its a string

File: client/src/components/organisms/TaskSlider/index.tsx (hunk 3 above)
Comment (rogeliojohnoliverio) [reply client/src/components/organisms/TaskSlider/index.tsx]: this is from query , its a string

**Review by jsvelte — APPROVED** [review]
# Approved 💯

File: client/src/components/organisms/TaskSlider/index.tsx (hunk 4 above)
Comment (rogeliojohnoliverio) [reply client/src/components/organisms/TaskSlider/index.tsx]: this is an event, which you said it is any at the moment

**Comment by ww-jermaine** [comment]
Approved 💯

**Review by abduljalilpalala — APPROVED** [review]
### Approved! ✔️

**Pull request merged**